src/AST/Class.cs

```diff
@@ -102,6 +102,9 @@ public class Class : DeclarationContext
         // True if the class represents a static class.
         public bool IsStatic;
 
+        // True if declared with 'struct' keyword
+        public bool DeclaredStruct;
```

Comment:
I have used 'DeclaredStruct' which is filled by AST just to check if while declaring this 'struct' keyword was used.
'IsStruct' is used to determine if the type follows all rules and should be marshalled to 'struct'...
.
Should I combine them both?